platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/recovery/internal/EventStreamBound.java

```diff
@@ -0,0 +1,287 @@
+    }
+
+    /**
+     * The string representation of the event stream bound.
+     *
+     * @return the string representation
+     */
+    @Override
+    public String toString() {
+        return "EventStreamBound{" + "round=" + round + ", timestamp=" + timestamp + '}';
+    }
+
+    /**
+     * Compares the detailed consensus event to this bound based on the consensus data in the event.
+     *
+     * @param event     the detailed consensus event to compare to
+     * @param boundType the type of bound to compare as
+     * @return a negative integer, zero, or a positive integer based on whether the consensus data is less, equal or
+     * greater than the bound.
+     */
+    public Integer compareTo(DetailedConsensusEvent event, BoundType boundType) {
+        return compareTo(event.getConsensusData(), boundType);
+    }
+
+    /**
+     * Compares the event implementation's consensus data to this bound.
+     *
+     * @param event     the event implementation to compare to
+     * @param boundType the type of bound to compare as
+     * @return a negative integer, zero, or a positive integer based on whether the consensus data is less, equal or
+     * greater than the bound.
+     */
+    public Integer compareTo(EventImpl event, BoundType boundType) {
+        return compareTo(event.getConsensusData(), boundType);
+    }
+
+    /**
+     * Compares the round received and consensus timestamp of the consensusData to the bound.  The comparison is made
+     * with the frame of reference of the consensus data. If the data is higher, a value greater than 0 is returned, if
+     * the data is lower, a valued less than 0 is returned, otherwise 0 is returned.
+     * <p>
+     * If the bound is a LOWER bound, if both the round and the timestamp are present in the bound then the consensus
+     * data is greater than bound if both the bound's round and the timestamp are lower than the consensus data round
+     * received and consensus timestamp, the consensus data is less than the bound if either the bound's round or the
+     * timestamp are strictly greater, and equal otherwise.   If only one of the timestamp or round is present, the
+     * normal comparison rules apply. If neither the timestamp nor round is present, the bound is an unbounded lower
+     * bound and the event is always greater than it.
+     * <p>
+     * If the bound is an UPPER bound, if both the round and the timestamp are present then the consensus data is less
+     * than the bound if both the bound's round and the timestamp are greater than the consensus data round received and
+     * consensus timestamp, the consensus data is greater than the bound if either bound's the round or the timestamp
+     * are strictly less, and equal otherwise.   If only one of the timestamp or round is present, the normal comparison
+     * rules apply. If neither the timestamp nor round is present, the bound is an unbounded upper bound and the event
+     * is always less than it.
```

Comment:
Is this class ever used as an upper bound (i.e. read the stream until we hit this boundary)? Our are we currently just using it as a lower bound (i.e. start reading the stream at this boundary)?

Reply:
Just as a lower bound.  I generalized to support use as an upper bound as well.

Reply:
I have removed the upper bound use.